tfhe/src/high_level_api/strings/ascii/mod.rs

```diff
@@ -0,0 +1,124 @@
+mod comp;
+mod contains;
+mod find;
+mod no_pattern;
+mod replace;
+mod strip;
+mod trim;
+
+use crate::high_level_api::details::MaybeCloned;
+use crate::prelude::{FheDecrypt, FheTryEncrypt, Tagged};
+use crate::strings::ciphertext::FheString;
+use crate::{ClientKey, Tag};
+pub use no_pattern::{FheStringIsEmpty, FheStringLen};
+
+pub enum EncryptableString<'a> {
+    NoPadding(&'a str),
+    WithPadding { str: &'a str, padding: u32 },
+}
+
+impl EncryptableString<'_> {
+    fn str_and_padding(&self) -> (&str, Option<u32>) {
+        match self {
+            EncryptableString::NoPadding(str) => (str, None),
+            EncryptableString::WithPadding { str, padding } => (str, Some(*padding)),
+        }
+    }
+}
+
+pub(crate) enum AsciiDevice {
+    Cpu(FheString),
+}
+
+impl From<FheString> for AsciiDevice {
+    fn from(value: FheString) -> Self {
+        Self::Cpu(value)
+    }
+}
+
+impl AsciiDevice {
+    pub fn on_cpu(&self) -> MaybeCloned<'_, FheString> {
+        match self {
+            Self::Cpu(cpu_string) => MaybeCloned::Borrowed(cpu_string),
+        }
+    }
+}
+
+pub struct FheAsciiString {
+    pub(crate) inner: AsciiDevice,
+    pub(crate) tag: Tag,
+}
+
+impl Tagged for FheAsciiString {
+    fn tag(&self) -> &Tag {
+        &self.tag
+    }
+
+    fn tag_mut(&mut self) -> &mut Tag {
+        &mut self.tag
+    }
+}
+
+impl FheAsciiString {
+    pub(crate) fn new(inner: impl Into<AsciiDevice>, tag: Tag) -> Self {
+        Self {
+            inner: inner.into(),
+            tag,
+        }
+    }
+
+    pub fn try_encrypt_with_padding(
+        str: impl AsRef<str>,
+        padding: u32,
+        client_key: &ClientKey,
+    ) -> crate::Result<Self> {
```

Comment:
Maybe it would be easier for the users to give the total size instead of the padding count here, so they don't have to compute it from the string size

Reply:
we could add a `try_encrypt_with_fixed_sized` which encrypts and padds (or trims) so that the str has the given size

Reply:
yes good idea !